src/manipulators/strict_null_checks_manipulator.ts

```diff
@@ -555,4 +535,55 @@ export class StrictNullChecksManipulator extends Manipulator {
       );
     }
   }
+
+  /**
+   * Parses through a list of types and removes unnecessary types caused by any and never.
+   * @param {string[]} types - List of types.
+   * @return {string[]} List of filtered types.
+   */
+  private filterUnnecessaryTypes(types: string[]): string[] {
+    return types.filter((type, index) => {
```

Comment:
let's not use filter, instead let's reconstruct the types.

you can do a pass on types and split to several groups:
single_type
single_type[]
single_type[][], (I think this is enough, and no need to triple "[]")

then in each group, just reduce to any or never if any or never appear.

Reply:
I agree that the current implementation is confusing, I rewrote it in a much more clearer way to not use filter anymore, we can also discuss this later today in our meeting.